security/user-last-seen.php

```diff
@@ -0,0 +1,273 @@
+				'meta_type'    => 'NUMERIC',
+				'meta_compare' => '<',
+				'count_total'  => true,
+			),
+		);
+		$count       = (int) $users_query->get_total();
+
+		$view = __( 'Blocked Users' );
+		if ( $count ) {
+			$url = add_query_arg( array(
+				'last_seen_filter'       => 'blocked',
+				'last_seen_filter_nonce' => wp_create_nonce( 'last_seen_filter' ),
+			) );
+
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$class = isset( $_GET['last_seen_filter'] ) ? 'current' : '';
+
+			$view = '<a class="' . esc_attr( $class ) . '" href="' . esc_url( $url ) . '">' . esc_html( $view ) . '</a>';
+		}
+		$views['blocked_users'] = $view . ' (' . $count . ')';
+
+		return $views;
+	}
+
+	public function last_seen_unblock_action() {
+		$admin_notices_hook_name = is_network_admin() ? 'network_admin_notices' : 'admin_notices';
+
+		if ( isset( $_GET['reset_last_seen_success'] ) && '1' === $_GET['reset_last_seen_success'] ) {
+			add_action( $admin_notices_hook_name, function() {
+				$class = 'notice notice-success is-dismissible';
+				$error = __( 'User unblocked.', 'inactive-account-lockdown' );
+
+				printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), esc_html( $error ) );
+			} );
+		}
+
+		if ( ! isset( $_GET['user_id'] ) || ! isset( $_GET['action'] ) || 'reset_last_seen' !== $_GET['action'] ) {
+			return;
+		}
+
+		$user_id = absint( $_GET['user_id'] );
+
+		$error = null;
+		if ( ! isset( $_GET['reset_last_seen_nonce'] ) || ! wp_verify_nonce( sanitize_text_field( $_GET['reset_last_seen_nonce'] ), 'reset_last_seen_action' ) ) {
+			$error = __( 'Unable to verify your request', 'inactive-account-lockdown' );
+		}
+
+		if ( ! get_userdata( $user_id ) ) {
+			$error = __( 'User not found.', 'inactive-account-lockdown' );
+		}
+
+		if ( ! current_user_can( 'edit_user', $user_id ) ) {
+			$error = __( 'You do not have permission to unblock this user.', 'inactive-account-lockdown' );
+		}
+
+		if ( ! $error && ! delete_user_meta( $user_id, self::LAST_SEEN_META_KEY ) ) {
+			$error = __( 'Unable to unblock user.', 'inactive-account-lockdown' );
+		}
+
+		if ( $error ) {
+			add_action( $admin_notices_hook_name, function() use ( $error ) {
+				$class = 'notice notice-error is-dismissible';
+
+				printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), esc_html( $error ) );
+			} );
+			return;
+		}
+
+		$url = remove_query_arg( array(
+			'action',
+			'user_id',
+			'reset_last_seen_nonce',
+		) );
+
+		$url = add_query_arg( array(
+			'reset_last_seen_success' => 1,
+		), $url );
+
+		wp_safe_redirect( $url );
+		exit();
+	}
+
+	public function register_release_date() {
+		if ( ! get_option( self::LAST_SEEN_RELEASE_DATE_TIMESTAMP_OPTION_KEY ) ) {
+			// Right after the first admin_init, set the release date timestamp
+			// to be used as a fallback for users that never logged in before.
+			add_option( self::LAST_SEEN_RELEASE_DATE_TIMESTAMP_OPTION_KEY, time() );
```

Comment:
Let's make this a non-autoloaded option, since it's not something needed on every request.

Reply:
TIL about autoloaded options!

Implemented on https://github.com/Automattic/vip-go-mu-plugins/pull/4765/commits/bb49b0cb066c6e26aff2f395733629ecd457c1fb